src/cgns_internals.c

```diff
@@ -49,6 +52,7 @@ extern int HDF5storage_type;
 int Idim;           /* current IndexDimension          */
 int Cdim;           /* current CellDimension           */
 int Pdim;           /* current PhysicalDimension           */
+cgns_zone *CurrentZonePtr; /* current Zone structure pointer */
```

Comment:
Due to High order specificities, It is required to acces the Element sections in the solution data size method. I did it to avoid modifying too deeply the existing methods. However, I could modify the `cgi_read_sol()` to pass the zone as argument ...